README.Rmd

```diff
@@ -27,11 +27,107 @@ to create .NET objects, access their fields, and call their methods.
 This package is based on the [rClr](https://github.com/rdotnet/rClr) package and
 utilizes some of its code base.
 
-## Installation
+The package utilizes two code bases - the R code which is exposed to the user, 
+and the C++/C# code that communicates with the .NET libraries. The end user of 
+the package will only interact with the R code, and pre-build C++/C# libraries are
+supplied with the package. See section [Installation](#installation) for instructions on how to install the package.
+
+Advanced users and C++/C# developers might want to build the C++/C# code from source.
+For this, follow the instructions in section [Build](#build).
+
+## Installation 
 
 ### Windows
 
-### Linux
+You will need the latest Microsoft Visual C++ Redistributable for Visual Studio 2015, 2017 and 2019 (you can get it from [here](https://support.microsoft.com/en-us/help/2977003/the-latest-supported-visual-c-downloads) 
+and .NET 8 runtime (you can get it from [here](https://dotnet.microsoft.com/download/dotnet/8.0/runtime)).
+
+You can install the package from GitHub with pre-build binaries by running:
+
+```{r, eval=FALSE}
+install.packages("remotes")
+remotes::install_github("Open-Systems-Pharmacology/rSharp@*release")
+```
+
+Get the latest development version with:
+
+```{r, eval=FALSE}
+remotes::install_github("Open-Systems-Pharmacology/rSharp")
+```
+
+Alternatively, download the [latest release](https://github.com/Open-Systems-Pharmacology/rSharp/releases) of the package and install it locally.
+
+### Ubuntu
+
+Run the following commands to install the required dependencies:
+
+```
+sudo apt-get install dotnet-runtime-8.0 libcurl4-openssl-dev libssl-dev libxml2-dev 
+sudo apt-get install libfontconfig1-dev libharfbuzz-dev libfribidi-dev
+sudo apt-get install libfreetype6-dev libpng-dev libtiff5-dev libjpeg-dev
+sudo ln -s /usr/lib/x86_64-linux-gnu/libdl.so.2 /usr/lib/x86_64-linux-gnu/libdl.so
+```
+Symlink libdl.so.2 to libdl.so
```

Comment:
What is this? Why?

Reply:
We need this due to a dependency in ```dynamic-interop-dll```. It's solved in a PR in our clone but all the dependencies need to be updated. I think we decided to pull everything into one solution to avoid excessive dependency management.

https://github.com/Open-Systems-Pharmacology/dynamic-interop-dll/pull/1

Added this issue

https://github.com/Open-Systems-Pharmacology/rSharp/issues/120